### PR TITLE
First draft of a Prometheus 2.0 Stats dashboard

### PR DESCRIPTION
**Pull request by cboggs**
Here's a first stab at a Prometheus 2.0 default dashboard, as per #9759.

### CODE REVIEW

**Comment by cboggs** [comment]
CLA is signed, just waiting for the box to flip to green. :-)

**Comment by bergquist** [comment]
Please add the email address you used in the git commit to your github account. That's how the CLA bot matches commit with the list of contributors.

**Comment by cboggs** [comment]
Ok, added and confirmed it seems. Sorry for the confusion - had an old (inaccessible) email address in my local config that I forgot about. :-)

I amended and tried to squash and force push to my branch, but it didn't get remove the old user. Can re-open if that works better?

**Comment by Gouthamve** [comment]
Hi, so I am just imported this dashboard and as someone who has not run Prometheus in serious environments in a long-time, I don't think I'm the right person either. A few comments though:

Not sure if its right, but importing it is throwing an error. Somehow the datasource is not being set right throwing: `Datasource named ${DS_LOCAL-P8S} was not found`

Also, the templating is broken. I think instead of picking `localhost`, it is picking `localhost:9090` which breaks the `$node.*` by not selecting `node_exporter` metrics. Also the templating assumes `node_exporter` is running and I am not sure including those metrics in a Prometheus default dashboard is right.

My small server's snapshot: https://snapshot.raintank.io/dashboard/snapshot/0mspq3nfItd0sNoZ7kZ03rIVUeKkLybO

File: public/app/plugins/datasource/prometheus/dashboards/prometheus_2_stats.json
Comment (Gouthamve) [inline public/app/plugins/datasource/prometheus/dashboards/prometheus_2_stats.json]: `$node` was used above.

File: public/app/plugins/datasource/prometheus/dashboards/prometheus_2_stats.json
Comment (Gouthamve) [inline public/app/plugins/datasource/prometheus/dashboards/prometheus_2_stats.json]: `_ftio`?

**Review by gouthamve — COMMENTED** [review]
Few minor nits

File: public/app/plugins/datasource/prometheus/dashboards/prometheus_2_stats.json
Comment (tinytub) [inline public/app/plugins/datasource/prometheus/dashboards/prometheus_2_stats.json]: node template maybe could use a regex like `"regex": "(.*):\\d+",`,
prometheus exporter path and node_exporter exporter path always with  diffrent port, will cause `instance=~'[[node]].*'` can not get right node address

**Review by tinytub — COMMENTED** [review]
after imported and try

File: public/app/plugins/datasource/prometheus/dashboards/prometheus_2_stats.json
Comment (cboggs) [reply public/app/plugins/datasource/prometheus/dashboards/prometheus_2_stats.json]: Oops, missed some copy/paste queries.

**Comment by cboggs** [comment]
Agreed that assuming node_exporter presence is probably not wise for a built-in. Will adjust to remove it and it should simplify things considerably.

**Comment by cboggs** [comment]
@gouthamve, @tinytub, I believe I've cleaned it up a bit. Should be working better now, re-tested with a clean local Prometheus + Grafana. Let me know if you see other changes to be made, thanks for the feedback!

**Comment by danielmarbach** [comment]
There is also https://grafana.com/dashboards/3662

**Comment by Gouthamve** [comment]
@cboggs This imports cleanly :) One nit:
* `scrape_duration_seconds` is the metric for scrape duration and not `prometheus_target_interval_length_seconds`

A suggestion:
* Showing the number of rule_iterations and missed_iterations would also help.

**Comment by cboggs** [comment]
@Gouthamve, done and done! Not sure where I picked up the `prometheus_target_interval_length_seconds` - good catch! Thanks for the inputs. :-)

**Comment by bergquist** [comment]
@Gouthamve @cboggs anything else you want to add before I merge this? :)

**Comment by cboggs** [comment]
@bergquist unless @Gouthamve has any suggestions, I'm happy with it. :-)

File: public/app/plugins/datasource/prometheus/dashboards/prometheus_2_stats.json
Comment (Gouthamve) [inline public/app/plugins/datasource/prometheus/dashboards/prometheus_2_stats.json]: Legend format wrong

File: public/app/plugins/datasource/prometheus/dashboards/prometheus_2_stats.json
Comment (Gouthamve) [inline public/app/plugins/datasource/prometheus/dashboards/prometheus_2_stats.json]: Can this be over all jobs? Like `topk(5, max(scrape_duration_seconds) by (job))`

If that is the case then legend should be: `{{job}}`

**Review by gouthamve — CHANGES_REQUESTED** [review]
Well, another minor nit with the legend format :P 

Other than that :+1: Once that is fixed, you can merge without waiting for my review :)

**Comment by cboggs** [comment]
Excellent, I didn't pay attention to the fact that `scrape_duration_seconds` was broken up per target. :+1: Updated @Gouthamve and @bergquist. 

Thanks again!

**Comment by bergquist** [comment]
Merged by https://github.com/grafana/grafana/commit/14b8bd35ed76f35bc4fe7596204cd9b30ef98d24

Not sure why github dont flag this as merged.

**Pull request closed without merging**
